From 71464db9c2492478f2b396d3954e71b38d28e8c3 Mon Sep 17 00:00:00 2001
From: Saurav Girepunje <saurav.girepunje@gmail.com>
Date: Mon, 7 Oct 2019 23:56:53 +0530
Subject: [PATCH] usb: core: devio.c: Fix assignment of 0/1 to bool variables
Git-commit: 71464db9c2492478f2b396d3954e71b38d28e8c3
References: CVE-2021-47582 bsc#1226559
Patch-mainline: v5.5-rc1

Use true/false for is_in bool type in function proc_do_submiturb.

Signed-off-by: Saurav Girepunje <saurav.girepunje@gmail.com>
Link: https://lore.kernel.org/r/20191007182649.GA7068@saurav
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Oliver Neukum <oneukum@suse.com>

---
 drivers/usb/core/devio.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/usb/core/devio.c
+++ b/drivers/usb/core/devio.c
@@ -1512,10 +1512,10 @@ static int proc_do_submiturb(struct usb_
 		uurb->buffer_length = le16_to_cpup(&dr->wLength);
 		uurb->buffer += 8;
 		if ((dr->bRequestType & USB_DIR_IN) && uurb->buffer_length) {
-			is_in = 1;
+			is_in = true;
 			uurb->endpoint |= USB_DIR_IN;
 		} else {
-			is_in = 0;
+			is_in = false;
 			uurb->endpoint &= ~USB_DIR_IN;
 		}
 		snoop(&ps->dev->dev, "control urb: bRequestType=%02x "
